File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Go/s612878490_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	firstLine := scanner.Text()

	parts := strings.SplitN(firstLine, " ", 2)
	N, _ := strconv.Atoi(parts[0])
	M, _ := strconv.Atoi(parts[1])

	shortage := make([]int, M)

	for i := 0; i < M; i++ {
		scanner.Scan()
		line := scanner.Text()
		Aparts := strings.SplitN(line, " ", 2)
		A, _ := strconv.Atoi(Aparts[0])
		if N > A {
			shortage[i] = N - A
		} else {
			shortage[i] = 0
		}
	}

	sort.Ints(shortage)

	ans := 0
	for i := 0; i < M-1; i++ {
		ans += shortage[i]
	}

	fmt.Println(ans)
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/C++/s612878490_commented.cpp
#include <iostream>
#include <algorithm>
#include <vector>

int main() {
    std::string line;
    std::getline(std::cin, line);
    int N, M;
    N = std::stoi(line.substr(0, line.find(' ')));
    M = std::stoi(line.substr(line.find(' ') + 1));

    std::vector<int> shortage(M);

    for (int i = 0; i < M; i++) {
        std::getline(std::cin, line);
        int A = std::stoi(line.substr(0, line.find(' ')));
        shortage[i] = N > A ? N - A : 0;
    }

    std::sort(shortage.begin(), shortage.end());

    int ans = 0;

    for (int i = 0; i < M - 1; i++) {
        ans += shortage[i];
    }

    std::cout << ans << std::endl;

    // End of Code
}
